wcfsetup/install/files/lib/data/user/UserProfileAction.class.php

```diff
@@ -609,6 +614,83 @@ public function validateUploadCoverPhoto()
         $uploadHandler->validateFiles(new UserCoverPhotoUploadFileValidationStrategy());
     }
 
+    /**
+     * Sets an avatar for a given user.
+     *
+     * @throws UserInputException If none or more than one user is given.
+     * @throws \InvalidArgumentException If the given avatar is not an image or is incorrect sized.
+     * @throws \RuntimeException If the avatar could not be saved for any reasons.
+     */
+    public function setAvatar(): array
```

Comment:
Move this method somewhere else, you are currently splitting `validateUploadCoverPhoto` and `uploadCoverPhoto` itself.